cfme/services/catalogs/catalog_item.py

```diff
@@ -145,17 +131,18 @@ def create(self):
         sel.wait_for_element(basic_info_form.name_text)
         catalog = fakeobject_or_object(self.catalog, "name", "Unassigned")
         dialog = fakeobject_or_object(self.dialog, "name", "No Dialog")
-        sel_provider = None
-        if self.item_type is not "Generic":
-            if self.provider.type_tclass is not "infra":
-                sel_provider = self.provider
+        # Provider is only in the create form for AnsibleTower type
+        select_provider = None
+        if not isinstance(self.provider, NoneType) \
+                and 'AnsbileTower' in [self.item_type, self.provider_type]:
```

Comment:
@sshveta can you check this out? I went through the 5.6 and 5.7 catalog item create forms, and in both cases the provider selection was _only_ present when Ansible Tower was selected as the type in the first part of the form.

The logic you had put in before was checking provider.type_tclass was not infra? What was the goal/intent there, what behavior in the UI were you trying to model? Thanks!